plugins/pow/pow.go

```diff
@@ -0,0 +1,81 @@
+package pow
+
+import (
+	"context"
+	"crypto"
+	"crypto/ed25519"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/iotaledger/goshimmer/packages/pow"
+	"github.com/iotaledger/goshimmer/plugins/config"
+	"github.com/iotaledger/hive.go/logger"
+	_ "golang.org/x/crypto/blake2b" // required by crypto.BLAKE2b_512
+)
+
```

Comment:
Consider making a constant denoting the amount of bytes the nonce takes then use that in the tests where you cut off message bytes to only the PoW relevant bytes.